include/deal.II/base/smartpointer.h

```diff
@@ -203,7 +203,7 @@ class SmartPointer
    * The Smartpointer is invalidated when the object pointed to is destroyed
    * or moved from.
    */
-  bool pointed_to_object_is_alive;
+  std::atomic<bool> pointed_to_object_is_alive;
```

Comment:
do you need a header file for that?

Reply:
`<atomic>` is included in `subscriptor.h`.